src/main/java/frc/robot/Constants.java

```diff
@@ -59,13 +62,27 @@ public class Constants {
                     MAX_ANGULAR_VELOCITY.in(Units.RotationsPerSecond),
                     MAX_ANGULAR_ACCELERATION.in(Units.RotationsPerSecond.per(Units.Second)));
 
-    public static final Pose2d[] OPTIMAL_POINTS_SHOOT = {
-        new Pose2d(1.97, 7.16, Rotation2d.fromDegrees(-161.57)),
-        new Pose2d(2.54, 3.06, Rotation2d.fromDegrees(-157.15)),
-        new Pose2d(4.16, 5.03, Rotation2d.fromDegrees(179.94))
-    };
-    public static final Pose2d[] OPTIMAL_POINTS_TRAP = {};
-    public static final Pose2d AMP_POSE = new Pose2d(0.0, 0.0, new Rotation2d(0, 0));
+    public static final List<Pose2d> OPTIMAL_POINTS_SHOOT_BLUE =
+            List.of(
+                    new Pose2d(1.97, 7.16, Rotation2d.fromDegrees(-161.57)),
+                    new Pose2d(2.54, 3.06, Rotation2d.fromDegrees(-157.15)),
+                    new Pose2d(4.16, 5.03, Rotation2d.fromDegrees(179.94)));
+    public static final List<Pose2d> OPTIMAL_POINTS_SHOOT_RED =
+            OPTIMAL_POINTS_SHOOT_BLUE.stream()
+                    .map(GeometryUtil::flipFieldPose)
+                    .collect(Collectors.toList());
+
+    public static final List<Pose2d> OPTIMAL_POINTS_TRAP_BLUE =
+            List.of(new Pose2d(0, 0, Rotation2d.fromDegrees(0)));
+    public static final List<Pose2d> OPTIMAL_POINTS_TRAP_RED =
+            OPTIMAL_POINTS_TRAP_BLUE.stream()
+                    .map(GeometryUtil::flipFieldPose)
+                    .collect(Collectors.toList());
+    public static final Translation2d AMP_POSE_BLUE = new Translation2d(1.85, 7.49);
+    public static final Translation2d AMP_POSE_RED = GeometryUtil.flipFieldPosition(AMP_POSE_BLUE);
+    public static final Rotation2d AMP_ROTATION_NORMAL = new Rotation2d(Math.toRadians(90));
+    public static final Rotation2d AMP_ROTATION_REVERSE = new Rotation2d(Math.toRadians(-90));
```

Comment:
Won't these be reversed when we're red?

Reply:
No, the field is Asymmetrical

Reply:
No, it's symmetrical

Reply:
![images](https://github.com/Galaxia5987/robot-2024/assets/68773850/94e0afc4-b7cf-4feb-9f10-ad71308ec6ba)
The rotation of the amp is the same for blue & red

Reply:
dffe6e

Reply:
It's never been like that. The odometry was always relative to the side we're on...

Reply:
Oh pathplanner works the other way. I guess we'll switch this year. I was right about the rotation though.

Reply:
But the measurement you get from AprilTags is global, not relative to your alliance. See also the warning at the end of this section: https://pathplanner.dev/pplib-build-an-auto.html#configure-autobuilder

Reply:
Also here: https://docs.wpilib.org/en/stable/docs/software/basic-programming/coordinate-system.html#always-blue-origin

Reply:
Reverted: 1ee9977

Reply:
Sorry for commenting after you, my network was slow and I didn't see your comment.
> I was right about the rotation though.

Why? Your pose is global, and both Amplifiers are on the same side.